gulpfile.js

```diff
@@ -0,0 +1,101 @@
+var _ = require('lodash');
+var gulp = require('gulp');
+var gutil = require('gulp-util');
+var ts = require('gulp-typescript');
+var tslint = require('gulp-tslint');
+var realTslint = require('tslint');
+var spawnSync = require('child_process').spawnSync;
+
+function build(src, dest) {
+  var tsProject = ts.createProject('configs/tsconfig-base.json', { declaration: true });
+  return gulp.src(src)
+    .pipe(tsProject())
+    .pipe(gulp.dest(dest));
+}
+
+gulp.task('build:capnp', function () {
+  return build('./packages/capnp-ts/src/**/*.ts', 'packages/capnp-ts/lib');
+});
+
+gulp.task('build:capnpc', function () {
+  // Stub, schema are not compiled yet
+  return build('./packages/capnpc-ts/src/**/*.ts', 'packages/capnpc-ts/lib');
+});
+
+gulp.task('build', ['build:capnp'/*, 'build:capnpc'*/]);
+
+function test(src, dest, coverage) {
+  var tsProject = ts.createProject('configs/tsconfig-base.json');
+  return gulp.src(src)
+    .pipe(tsProject())
+    .pipe(gulp.dest(dest))
+    .pipe(gutil.buffer(function (err, files) {
+      var options = ['-J'];
+      if (coverage) {
+        options.push('--cov', '--nyc-arg=-x=**/lib-test/**/*');
+      }
+      // lodash is used for Node 4 compatibility
+      var result = spawnSync('./node_modules/.bin/tap', options.concat(_.filter(_.map(files, function (file) {
```

Comment:
Yeah, that list is a little misleading; `Array.prototype.filter` & `Array.prototype.map` are part of the ES5 standard so totally fine on node 4 (I even tried it on a node 4 repl just to be sure!).